apps/web/src/pages/templates/workflow/DelayMetadata.tsx

```diff
@@ -0,0 +1,68 @@
+import { Grid, InputWrapper } from '@mantine/core';
+import { DigestUnitEnum } from '@novu/shared';
+import { Controller, useFormContext } from 'react-hook-form';
+import { Input, Select } from '../../../design-system';
+import { inputStyles } from '../../../design-system/config/inputs.styles';
+import { useEnvController } from '../../../store/use-env-controller';
+
+export const DelayMetadata = ({ control, index }) => {
+  const { readonly } = useEnvController();
+  const {
+    formState: { errors },
+  } = useFormContext();
+
+  return (
+    <>
+      <InputWrapper label="Time Interval" description="Once triggered, for how long should delay" styles={inputStyles}>
+        <Grid
+          sx={{
+            marginBottom: '2px',
+          }}
+        >
+          <Grid.Col span={4}>
+            <Controller
+              control={control}
+              name={`steps.${index}.metadata.amount`}
+              render={({ field }) => {
+                return (
+                  <Input
+                    {...field}
+                    error={errors?.steps ? errors.steps[index]?.metadata?.amount?.message : undefined}
+                    min={0}
+                    max={100}
+                    type="number"
+                    data-test-id="time-amount"
+                    placeholder="0"
+                  />
+                );
+              }}
+            />
+          </Grid.Col>
+          <Grid.Col span={8}>
+            <Controller
+              control={control}
+              name={`steps.${index}.metadata.unit`}
+              render={({ field }) => {
+                return (
+                  <Select
+                    disabled={readonly}
+                    error={errors?.steps ? errors.steps[index]?.metadata?.unit?.message : undefined}
+                    placeholder="Interval"
+                    data={[
+                      { value: DigestUnitEnum.SECONDS, label: 'Seconds' },
+                      { value: DigestUnitEnum.MINUTES, label: 'Minutes' },
+                      { value: DigestUnitEnum.HOURS, label: 'Hours' },
+                      { value: DigestUnitEnum.DAYS, label: 'Days' },
+                    ]}
+                    data-test-id="time-unit"
+                    {...field}
+                  />
```

Comment:
Is this also reused with the digest step? I think we can make a dumb component out of it. Wdyt?

Reply:
Yes, I will. I will also do some more refactoring to the WorkflowEditorPage to avoid duplicate code, something more general for step vs action step.